README.adoc

```diff
@@ -1,58 +1,110 @@
 == ALMighty Core
 
-image:https://goreportcard.com/badge/github.com/almighty/almighty-core[Go Report Card, link="https://goreportcard.com/report/github.com/almighty/almighty-core"] 
+image:https://goreportcard.com/badge/github.com/almighty/almighty-core[Go Report Card, link="https://goreportcard.com/report/github.com/almighty/almighty-core"]
 image:https://godoc.org/github.com/almighty/almighty-core?status.png[GoDoc,link="https://godoc.org/github.com/almighty/almighty-core"]
 
-Assuming you have Go installed and configured (have $GOPATH setup) here is how to build.
+== Building from source [[building]]
 
-Checking out the code:
+=== Prerequisites [[prerequisites]]
+
+You need to install `go` (>= v1.5) and `git`.
+
+==== Check your Go version [[check-go-version]]
+
+Run the following command to find out your Go version.
 
 ----
-git clone https://github.com/almighty/almighty-core $GOPATH/src/github.com/almighty/almighty-core
+$ go version
 ----
 
-Fetch necessary dependencies:
+If you are using Go `1.5` ensure the environment variable `GO15VENDOREXPERIMENT`
+is set, for example by running `export GO15VENDOREXPERIMENT=1`.
+In Go `1.6` it is enabled by default and in Go `1.7` it is always enabled
+without the ability to turn it off.
+
+If set, the `GO15VENDOREXPERIMENT` variable tells `go` to look for dependencies
+not only in the `GOPATH` (as usual) but also in a `vendor` directory located in
+the source directory.
+
+See <<fetch-dependencies>> to see an explanaition on how we deal with
+dependencies.
+
+==== Install glide [[glide-setup]]
+
+This project uses link:https://glide.sh/[glide]as a package manager for Go.
+
+To install glide, go to the
+link:https://github.com/Masterminds/glide/releases[release page] and download
+the newest binary for your operating system.
+
+Unpack the archive that you've downloaded and place the `glide` executable
+somewhere so that it is in your `PATH`.
+
+To check if everything is working, type `glide --version` in a terminal.
 
 ----
-cd $GOPATH/src/github.com/almighty/almighty-core
-make deps
+$ glide --version
+glide version v0.11.0
 ----
 
-=== Development
+Try to use at least version `0.11.0` as we haven't tested another version.
+
+=== Get the code [[get-the-code]]
 
-Only files ./*.go and ./design/*.go should be edited.
-Everything else, ./app, ./client, ./models, ./swagger, is generated.
+Assuming you have Go installed and configured (have `$GOPATH` setup) here is
+how to build.
 
-=== Build instructions
+Check out the code
 
-==== Generate GOA sources
-[source, console]
 ----
-make generate
+$ mkdir -p $GOPATH/src/github.com/almighty
```

Comment:
this `mkdir` is afaik redundant. `git clone` will create the parent dirs if necessary afaics.